src/adera/Machines/RCSController.cpp

```diff
@@ -127,16 +128,22 @@ void SysMachineRCSController::update_controls(ActiveScene& rScene)
         using wiretype::AttitudeControl;
         using Magnum::Vector3;
 
+        auto physComps = SysPhysics_t::try_get_or_find_rigidbody_parent(
+            rScene, ent, machine.m_rigidBody);
+        if (!physComps.first || !physComps.second) { continue; }
+
+        ACompRigidBody_t *compRb = physComps.first;
+        ACompTransform *compTf = physComps.second;
+
         if (WireData *gimbal = machine.m_wiCommandOrient.connected_value())
         {
             AttitudeControl *attCtrl = std::get_if<AttitudeControl>(gimbal);
 
             // TODO: RCS translation is not currently implemented, only rotation
             Vector3 commandTransl = Vector3{0.0f};
             Vector3 commandRot = attCtrl->m_attitude;
-            // HACK: hard-coded ship CoM, need to compute programmatically
-            Vector3 tmpOset{0.0f, 0.0f, -3.12f};
-            Vector3 thrusterPos = transform.m_transform.translation() - tmpOset;
+            Vector3 thrusterPos =
+                transform.m_transform.translation() - compRb->m_centerOfMassOffset;
```

Comment:
Similarly, I'm suspicious of this `transform` value from the Hierarchy. I think it's valid for now since the hierarchy structure of the vehicles is flat, but I don't think it holds in general. I think `find_rigidbody_ancestor()` is necessary to traverse the chain like in MachineRocket.